GPU/Common/FlatObject.h

```diff
@@ -437,7 +447,51 @@ inline void FlatObject::setFutureBufferAddress(char* futureFlatBufferPtr)
   mFlatBufferContainer = nullptr;
 }
 
-#ifndef GPUCA_GPUCODE // code invisible on GPU
+#if !defined(GPUCA_GPUCODE) // code invisible on GPU
+
+template <class T, class TFile>
+inline int FlatObject::writeToFile(T& obj, TFile& outf, const char* name)
```

Comment:
Why the TFile is a template parameter?

Reply:
HI Ruben, 

When I include  "TFile.h", or just "TObject.h" from FlatObject.h, the GPUTracker code doesn't compile with this error:

DEBUG:O2:O2:0:                  from /home/alidock/.sw/SOURCES/O2/v1.2.0/0/GPU/GPUTracking/SliceTracker/GPUTPCTracker.cxx:21:
DEBUG:O2:O2:0: /home/alidock/.sw/slc7_x86-64/ROOT/v6-20-02-alice3-1/include/TObject.h:179:40: error: declaration does not declare anything [-fpermissive]
DEBUG:O2:O2:0:     __attribute__((format(printf, 3, 4)))   /* 1 is the this pointer */
DEBUG:O2:O2:0:                                         ^

I didn't manage to fix this problem.

Reply:
Did you try to include it in the ``#if !defined(GPUCA_GPUCODE)`` guards?

Reply:
yes, it didn't help. As I remember, 

#if !defined(GPUCA_GPUCODE) && !defined(GPUCA_STANDALONE)

doesn't help either.

Reply:
Actually, for the ``MatLayerCylSet`` @davidrohr suggested to use ``#ifndef GPUCA_ALIGPUCODE``, could you try with that?

Reply:
It works, thanks! 
I need to verify that it won't break standalone and AliRoot compilations.

Reply:
@shahor02 ,  it works. 
Should I replace GPUCA_GPUCODE to GPUCA_ALIGPUCODE everywhere?